### ISSUE TITLE
java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.String

### ISSUE THREAD

**Issue by chiragpurohit71085** [invertase/react-native-firebase-docs#221]
I am trying to integrate firebase.auth().verifyPhoneNumber("+911234567890"). In ios it works properly and I am receiving firebase.auth.PhoneAuthState.CODE_SENT. In android app crashes and gives me  java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.String.

I have used with my actual number. 

Below is the code I have used

````

firebase.auth().verifyPhoneNumber(phoneNumber).on('state_changed', (phoneAuthSnapshot) => {
	switch (phoneAuthSnapshot.state) {
		// ------------------------
		//  IOS AND ANDROID EVENTS
		// ------------------------
		case firebase.auth.PhoneAuthState.CODE_SENT: // or 'sent'
			console.log('code sent');
			// on ios this is the final phone auth state event you'd receive
			// so you'd then ask for user input of the code and build a credential from it
			// as demonstrated in the `signInWithPhoneNumber` example above
			break;
		case firebase.auth.PhoneAuthState.ERROR: // or 'error'
			console.log('verification error');
			console.log(phoneAuthSnapshot.error);
			break;

		// ---------------------
		// ANDROID ONLY EVENTS
		// ---------------------
		case firebase.auth.PhoneAuthState.AUTO_VERIFY_TIMEOUT: // or 'timeout'
			console.log('auto verify on android timed out');
			// proceed with your manual code input flow, same as you would do in
			// CODE_SENT if you were on IOS
			break;
		case firebase.auth.PhoneAuthState.AUTO_VERIFIED: // or 'verified'
			console.log('auto verified on android');
			console.log(phoneAuthSnapshot);
			break;
	}
}, (error) => {
	// optionalErrorCb would be same logic as the ERROR case above,  if you've already handed
	// the ERROR case in the above observer then there's no need to handle it here
	console.log(error);
	// verificationId is attached to error if required
	console.log(error.verificationId);
}, (phoneAuthSnapshot) => {
	// optionalCompleteCb would be same logic as the AUTO_VERIFIED/CODE_SENT switch cases above
	// depending on the platform. If you've already handled those cases in the observer then
	// there's absolutely no need to handle it here.

	// Platform specific logic:
	// - if this is on IOS then phoneAuthSnapshot.code will always be null
	// - if ANDROID auto verified the sms code then phoneAuthSnapshot.code will contain the verified sms code
	//   and there'd be no need to ask for user input of the code - proceed to credential creating logic
	// - if ANDROID auto verify timed out then phoneAuthSnapshot.code would be null, just like ios, you'd
	//   continue with user input logic.
	console.log(phoneAuthSnapshot);
});



````

Below is the stack trace I am receiving.

````

E/AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
    Process: com.patientcircle.local, PID: 12105
    java.lang.ClassCastException: java.lang.Double cannot be cast to java.lang.String
        at com.facebook.react.bridge.ReadableNativeArray.getString(ReadableNativeArray.java:101)
        at com.facebook.react.bridge.JavaMethodWrapper$5.extractArgument(JavaMethodWrapper.java:73)
        at com.facebook.react.bridge.JavaMethodWrapper$5.extractArgument(JavaMethodWrapper.java:69)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:359)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
        at android.os.Looper.loop(Looper.java:193)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
        at java.lang.Thread.run(Thread.java:764)

````

**Comment by NaveenS18** [comment]
**Based On my Understanding:**
firebase.auth().verifyPhoneNumber("+911234567890").. This is the issue. You cannot assign double to a String. Java doesn't allow you to perform before you convert it.

**Solution:**
Get the input as a string and then pass it to firebase.auth().verifyPhoneNumber(**your String Variable**).

**Comment by mikehardy** [comment]
You can force it in the verify call by maybe doing `phoneNumber + ""`)

**Comment by chiragpurohit71085** [comment]
HI thanks for reply. I have tried booth ways, but still app is crashing. Just for note I am using typescript

````
console.log("phoneNumber new auth - ", this.state.phoneNumber);

let phoneNumberWithCountryCode: string = this.state.countryCode + "" + this.state.phoneNumber;
firebase.auth().verifyPhoneNumber(phoneNumberWithCountryCode+"").on('state_changed', (phoneAuthSnapshot) => {
.......

````

**Comment by chiragpurohit71085** [comment]
I also tried as per given example

     * ```js
     * firebase.auth().verifyPhoneNumber('+4423456789', )
     *  .on('state_changed', (phoneAuthSnapshot) => {
     *    console.log('Snapshot state: ', phoneAuthSnapshot.state);
     *  });
     * ```

**Comment by chiragpurohit71085** [comment]
Is there any suggestions here?

**Comment by mikehardy** [comment]
I thought this was fixed? https://github.com/invertase/react-native-firebase/commit/a2a296e39373137b94e21b653312d7904c36ac78

**Comment by Salakar** [comment]
> I thought this was fixed? [invertase/react-native-firebase@a2a296e](https://github.com/invertase/react-native-firebase/commit/a2a296e39373137b94e21b653312d7904c36ac78)

It sure is in v6.0.1! Must have missed this duplicate when closing the others. There's another issue thats come from this but tracking in a different issue. So will close this one.

**Comment by chiragpurohit71085** [comment]
Okay thanks. I will track other
